### PR TITLE
Dockerfile

### PR DESCRIPTION
**Pull request by bux578**
**When merged this pull request will:**
- Will add a Dockerfile to run jekyll in a container. This will allow you to NOT install all jekyll requirements on your PC.

_Note: this might need improvements with volumes etc..._

### CODE REVIEW

File: docs/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+# Run:
+# docker run -p 4000:4000 -it ace3mod/jekyll
+
+FROM starefossen/github-pages:onbuild
+MAINTAINER bux
+
+COPY . /usr/src/app
+WORKDIR /usr/src/app
+
+RUN gem install --no-document \
```
Comment (Glowbal) [inline docs/Dockerfile]: This seems unncessary. Doesn't the base image already cover this?

File: docs/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+# docker build -t ace3mod/jekyll .
+#
+# Run:
+# docker run -p 4000:4000 -it ace3mod/jekyll
+
+FROM starefossen/github-pages:onbuild
+MAINTAINER bux
+
+COPY . /usr/src/app
+WORKDIR /usr/src/app
```
Comment (Glowbal) [inline docs/Dockerfile]: Already done in the base image

File: docs/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+RUN gem install --no-document \
+  github-pages:91 \
+  jekyll-github-metadata
+
+ENTRYPOINT ["jekyll", "serve", "--future", "--incremental", "--config", "_config_dev.yml", "-H", "0.0.0.0", "-P", "4000"]
```
Comment (Glowbal) [inline docs/Dockerfile]: Use `CMD` instead. No reason to use `ENTRYPOINT`.

File: docs/Dockerfile (hunk 1 above)
Comment (Glowbal) [inline docs/Dockerfile]: Ah, you use the `onbuild` image for this. Probably want to switch to the `latest` instead?

File: docs/Dockerfile (hunk 2 above)
Comment (Glowbal) [inline docs/Dockerfile]: same as below, switch to latest, this is already done.

If not, the `COPY` on line 10 is unncessary.

File: docs/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Build:
+# docker build -t ace3mod/jekyll .
+#
+# Run:
+# docker run -p 4000:4000 -it ace3mod/jekyll
+
+FROM starefossen/github-pages:onbuild
```
Comment (Glowbal) [inline docs/Dockerfile]: I think using `latest` is probably a better solution for us.

**Comment by Glowbal** [comment]
Suggestion for the compose file:

```yml
version: '2'

services:
  docs:
    build: .
    restart: always
    ports:
      - "4000:4000"
    volumes:
      - .:/usr/src/app:rw

```

Note, probably double check the volume, I'm not sure if `.:/usr/src/app` works. Maybe `"$PWD":/usr/src/app` would be the better option.

File: docs/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# Build:
+# docker build -t ace3mod/jekyll .
+#
+# Run:
+# docker run -p 4000:4000 -it ace3mod/jekyll
+
+FROM starefossen/github-pages:latest
+MAINTAINER bux
+
+VOLUME "/usr/src/app"
```
Comment (Glowbal) [inline docs/Dockerfile]: Should we also copy the contents of `.` to `/usr/src/app` ? Means you won't have too mount in a volume always, but still can.

**Comment by bux578** [comment]
@Glowbal Should be good to go now.

**Comment by bux578** [comment]
👍

**Pull request merged**